lib/usbd/usbd_transfer.c

```diff
@@ -897,7 +897,7 @@ void *usbd_urb_get_buffer_pointer(usbd_device *dev, usbd_urb *urb, size_t len)
 		return transfer->buffer;
 	}
 
-	return transfer->buffer + transfer->transferred;
+	return (void*)((uintptr_t)transfer->buffer + transfer->transferred);
```

Comment:
isnt `transfer->buffer + transfer->transferred` pointer arithmetic (valid in c).
what is the reason of writing it in a more obsured way?